src/sacn/source.c

```diff
@@ -152,66 +381,228 @@ void sacn_source_destroy(sacn_source_t handle)
  * @return #kEtcPalErrInvalid: Invalid parameter provided.
  * @return #kEtcPalErrNotInit: Module not initialized.
  * @return #kEtcPalErrExists: Universe given was already added to this source.
- * @return #kEtcPalErrNotFound: Handle does not correspond to a valid source, or a network interface ID given was not
- * found on the system.
+ * @return #kEtcPalErrNotFound: Handle does not correspond to a valid source.
  * @return #kEtcPalErrNoMem: No room to allocate additional universe.
  * @return #kEtcPalErrSys: An internal library or system call error occurred.
  */
 etcpal_error_t sacn_source_add_universe(sacn_source_t handle, const SacnSourceUniverseConfig* config,
                                         SacnMcastInterface* netints, size_t num_netints)
 {
-  ETCPAL_UNUSED_ARG(handle);
-  ETCPAL_UNUSED_ARG(config);
-  ETCPAL_UNUSED_ARG(netints);
-  ETCPAL_UNUSED_ARG(num_netints);
-  return kEtcPalErrNotImpl;
+#if !SACN_SOURCE_ENABLED
+  return kEtcPalErrNotInit;
+#endif
+
+  etcpal_error_t result = kEtcPalErrOk;
+
+  // Verify module initialized.
+  if (!sacn_initialized())
+    result = kEtcPalErrNotInit;
+
+  // Check for invalid arguments.
+  if (result == kEtcPalErrOk)
+  {
+    if (handle == SACN_SOURCE_INVALID)
```

Comment:
Another if block to combine.

Reply:
Resolved in https://github.com/ETCLabs/sACN/pull/32/commits/4338ffdded0288b92cc19437c16825145d3966f0.